### PR TITLE
Dynamic partition

### PR DESCRIPTION
**Pull request by moezAhmad**

### CODE REVIEW

**Comment by moezAhmad** [comment]
#16085

File: ivy/functional/frontends/tensorflow/general_functions.py
Comment (theRealBird) [inline ivy/functional/frontends/tensorflow/general_functions.py:1]: Hi @moezAhmad, thank you for your contribution!

How are we accounting for this requirement ```data.shape``` must start with ```partitions.shape``` in the code implementation?

Comment (moezAhmad) [reply ivy/functional/frontends/tensorflow/general_functions.py:1]: @theRealBird, I'm grateful for your time in reviewing my Pull Request. Apologies for my delayed response, as I was not available online last week. In the meantime, I've updated the code to include checks for data and partitions Arrays. Unlike the TensorFlow implementation, which accepts multi-dimensional partitions, the current dynamic_partition function in our code is designed to work with 1D partition arrays.

While I feel the function dynamic_partition is a bit complex, I'm also considering the potential benefits of introducing it as part of the Ivy API in the initial phase, allowing for straightforward integration from the front end. I'd appreciate any guidance or feedback you could provide to further enhance this implementation.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

File: ivy/functional/frontends/tensorflow/general_functions.py
Comment (moezAhmad) [reply ivy/functional/frontends/tensorflow/general_functions.py:1]: @theRealBird  Thank you for your reply
Following the guide, I have created a new issue #17468  related to dynamic_partition.

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Pull request closed without merging**